src/_pytest/logging.py

```diff
@@ -72,24 +72,62 @@ class PercentStyleMultiline(logging.PercentStyle):
     formats the message as if each line were logged separately.
     """
 
+    def __init__(self, fmt, auto_indent):
+        super().__init__(fmt)
+        self._auto_indent = auto_indent
+
     @staticmethod
     def _update_message(record_dict, message):
         tmp = record_dict.copy()
         tmp["message"] = message
         return tmp
 
+    def _get_auto_indent(self, record):
```

Comment:
Can we add a docstring here please? It is not entirely obvious all that's being done here.

I also suggest turning this into a function:

```python
def get_auto_indent(record, default_auto_indent) -> int:
```

Where `default_auto_indent` is `self._auto_indent` at the call site.

Then we can unittest this function in isolation and account for all the branches and cases it handles. 👍

Reply:
Docstring: yes, I'll create that now.
Altering the method definition: Yes, I can make that static. Do you feel it is important to make the method public? To me it feels like it is private, but I don't know your views on calling private methods from the unittests.

Reply:
With respect to the docstring, you indicated that the purpose for it is that "It is not entirely obvious all that's being done here." As I began to write it, it started to seem wordy, because I'm trying to address your concern. Do you really want something detailed?

Reply:
More about the overall purpose of the method

> Altering the method definition:

I meant to make it a function; the fact that it is public is not really relevant, because the module itself is not public (but it is OK to import the private module on our own test suite).